packages/dev-server/src/MetroDevServer.ts

```diff
@@ -0,0 +1,45 @@
+import { getConfig, resolveModule } from '@expo/config';
+import { createDevServerMiddleware } from '@react-native-community/cli-server-api';
+import Log from '@expo/bunyan';
+import * as ExpoMetroConfig from '@expo/metro-config';
+import bodyParser from 'body-parser';
+
+import clientLogsMiddleware from './middleware/clientLogsMiddleware';
+import LogReporter from './LogReporter';
+
+export type MetroDevServerOptions = ExpoMetroConfig.LoadOptions & { logger: Log };
+
+export async function runMetroDevServerAsync(projectRoot: string, options: MetroDevServerOptions) {
+  const { exp } = getConfig(projectRoot);
```

Comment:
```suggestion
  const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
```
It'd be nice to change how we do `nodeModulesPath` so we can avoid having to read the config every time we want to resolve a string that rarely exists.

Reply:
Can you explain why `skipSDKVersionRequirement` is needed? Given that most call sites of `getConfig` pass in `skipSDKVersionRequirement: true`, maybe it should be the default? Then in those cases where it's actually needed we could call `getExpoSDKVersion(projectRoot, exp)`, which throws for missing `sdkVersion`? I think this would be a cleaner API, since normally you just want the config without caring too much about how the `sdkVersion` is treated.